libraries/standard/mqtt/utest/mqtt_utest.c

```diff
@@ -354,10 +362,12 @@ static void setupSubscriptionInfo( MQTTSubscribeInfo_t * pSubscribeInfo )
 static void expectProcessLoopCalls( MQTTContext_t * const pContext,
                                     MQTTStatus_t deserializeStatus,
                                     MQTTPublishState_t stateAfterDeserialize,
+                                    MQTTStatus_t updateStateStatus,
                                     MQTTStatus_t serializeStatus,
                                     MQTTPublishState_t stateAfterSerialize,
                                     MQTTStatus_t processLoopStatus,
-                                    bool incomingPublish )
+                                    bool incomingPublish,
+                                    MQTTPublishInfo_t * pPubInfo )
```

Comment:
The parameter list for this function is getting pretty large, and I don't see any comments describing each parameter either. At this point - since a lot of these parameters are related to each other - would it be more maintainable to create a struct to encapsulate most of these parameters and pass that in instead? I won't block the PR for this though

Reply:
Hi, I had a different set of opinions about the function we use. I see a few problems with the function already.
1. I think the readability of the tests are not great with the function. I have to spent a good amount of time to understand what the function is doing and spent another good amount time to write any tests for ProcessLoop. I think if we are not the developer of the function, the readability aspect is really poor.
2. Debugging issues - debugging was very difficult for any failure as the line number and the expected and actual values are not really pointing to the test failed in the group, Tests need to be rerun again with logs enabled or enabling tests one by one in the group. I think getting rid of the function itself will be a good change.

Anyway those are not in the scope of the PR, I think we would need to come back to it.